android/src/main/java/com/reactnativegeth/RNGethModule.java

```diff
@@ -81,30 +82,37 @@ public String getName() {
      */
     @ReactMethod
     public void nodeConfig(ReadableMap config, Promise promise) {
+        if (gethHolder.getNode() != null) {
+          Log.w(TAG, "RNGeth already has a node, skipping creation of a new one");
+          promise.resolve(true);
+          return;
+        }
+
         try {
-            NodeConfig nc = GethHolder.getNodeConfig();
+            Log.i(TAG, "Configuring node config");
+            NodeConfig nc = gethHolder.getNodeConfig();
             String nodeDir = ETH_DIR;
             String keyStoreDir = KEY_STORE_DIR;
             if (config.hasKey("enodes"))
-                GethHolder.writeStaticNodesFile(config.getString("enodes"));
+                gethHolder.writeStaticNodesFile(config.getString("enodes"));
             if (config.hasKey("networkID")) nc.setEthereumNetworkID(config.getInt("networkID"));
             if (config.hasKey("maxPeers")) nc.setMaxPeers(config.getInt("maxPeers"));
             if (config.hasKey("genesis")) nc.setEthereumGenesis(config.getString("genesis"));
             if (config.hasKey("nodeDir")) nodeDir = config.getString("nodeDir");
             if (config.hasKey("keyStoreDir")) keyStoreDir = config.getString("keyStoreDir");
             if (config.hasKey("syncMode")) nc.setSyncMode(config.getInt("syncMode"));
             if (config.hasKey("useLightweightKDF")) nc.setUseLightweightKDF(config.getBoolean("useLightweightKDF"));
-            if (config.hasKey("noDiscovery")) nc.setNoDiscovery(config.getBoolean("noDiscovery"));
-            if (config.hasKey("bootnodeEnodes")) {
-              ReadableArray bootnodeEnodes = config.getArray("bootnodeEnodes");
-              int enodesSize = bootnodeEnodes.size();
-              Enodes enodes = new Enodes(enodesSize);
-              for (int i = 0; i < enodesSize; i++) {
-                Enode enode = new Enode(bootnodeEnodes.getString(i));
-                enodes.set(i, enode);
-              }
-              nc.setBootstrapNodes(enodes);
-            }
+            // if (config.hasKey("noDiscovery")) nc.setNoDiscovery(config.getBoolean("noDiscovery"));
```

Comment:
@tkporter I had to comment our your recent changes to get this to compile. Do your changes need to be coordinated with some Geth changes?

Reply:
Ah yeah they do, I haven’t merged it yet because I wanted to coordinate this repo/geth/monorepo and I haven’t had the time, so this is totally fine